Libplanet.Tests/AddressTest.cs

```diff
@@ -46,6 +46,48 @@ public void DerivingConstructor()
             );
         }
 
+        [Fact]
+        public void HexAddressConstructor()
+        {
+            Assert.Equal(
+                new Address(
+                    new byte[20]
+                    {
+                        0x45, 0xa2, 0x21, 0x87, 0xe2, 0xd8, 0x85, 0x0b, 0xb3,
+                        0x57, 0x88, 0x69, 0x58, 0xbc, 0x3e, 0x85, 0x60, 0x92,
+                        0x9c, 0xcc,
+                    }
+                ),
+                new Address(
+                    "45a22187e2d8850bb357886958bc3e8560929ccc"
+                )
+            );
+
+            var address = new Address(
+                "45a22187e2d8850bb357886958bc3e8560929ccc"
+            );
+            Assert.Equal(
+                "45a22187e2D8850bb357886958bC3E8560929ccc",
+                address.ToHex()
+            );
+        }
+
+        [Fact]
+        public void HexAddressConstructorOnlyTakesASCIICharacters()
+        {
+            Assert.Throws<ArgumentException>(
+                () => new Address("45a22187e2d8850bb357886958BC3E8560929£한글")
```

Comment:
Even if a string consists of only ASCII characters, it should throw an exception unless it only consists of `[A-Fa-f0-9]`.  It also must not be longer nor shorter than 40 characters.  Could we add tests for those three cases?

- 40 ASCII characters which contain any `[^A-Fa-f0-9]` character
- 41 (or longer) of valid hexadecimal characters
- 39 (or shorter) of valid hexadecimal characters

Reply:
That would be good, I'll add the test cases.